Hazel/Dtls/DtlsConnectionListener.cs

```diff
@@ -1331,8 +1341,10 @@ protected override void QueueRawData(ByteSpan span, IPEndPoint remoteEndPoint)
                         ref outgoingRecord
                     );
 
-                    base.QueueRawData(packet, remoteEndPoint);
+                    queuedSpan.Recycle();
+                    base.QueueRawData(buffer, remoteEndPoint);
                 }
+
                 peer.QueuedApplicationDataMessage.Clear();
```

Comment:
For the Clear(), I think we'd need to recycle all these objects before Clearing?

Reply:
Refactored this to be a real queue so we feel more confident that its contents are always recycled.